tests/unit/interpolation_test.py

```diff
@@ -80,7 +80,7 @@ def binary_search_test():
 
     # Test the case with needle outside the haystack range
     assert binary_find(haystack, -1.) == (0, 0)
-    assert binary_find(haystack, 990.) == (98, 98)
+    assert binary_find(haystack, 98.) == (98, 98)
```

Comment:
You cannot arbitrarily change a test case such that it "works" :) 
The argument has to be 990 here, because for some reason for the code to function, the binary search has to be incorrect in this corner case. We are looking into this now. 

To be fair, I should have left a comment and probably marked the test case a "expected fail".